processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -594,6 +595,9 @@ public QueryableIndex load(File inDir, ObjectMapper mapper) throws IOException
       Map<String, Column> columns = Maps.newHashMap();
 
       for (String columnName : cols) {
+        if (Strings.isNullOrEmpty(columnName)) {
```

Comment:
I'm not sure this is the right thing to do, conceptually. Maybe log a warning, at least.

Reply:
Thanks! logged warning

Reply:
Please include `inDir` in this warning, it will help the user track down the specific malformed segment.